notebook/notebookapp.py

```diff
@@ -1908,13 +1923,12 @@ def shutdown_no_activity(self):
         if len(km) != 0:
             return   # Kernels still running
 
-        try:
-            term_mgr = self.web_app.settings['terminal_manager']
-        except KeyError:
-            pass  # Terminals not enabled
-        else:
-            if term_mgr.terminals:
-                return   # Terminals still running
+        if not self.terminals_in_use:
+            return
```

Comment:
This doesn't look quite right. This method is to shut down the server after a certain time with no kernels/terminals. Terminals not in use means no terminals running, but this early return prevents it from ever shutting down.

Reply:
:smile: Looks like we could do with another pass at naming, as you alluded to in another comment.  I was coming to same conclusion about [`terminals_in_use`](https://github.com/kevin-bates/notebook/blob/detect-missing-terminado/notebook/notebookapp.py#L1481-L1485) yesterday (while doing yard work).  I think a switch (alignment) to `terminals_available` might just be the best name:
a. `terminado_available` = Terminado package is installed
b. `terminals_enabled` = User has condoned the use of terminals (independent of terminado's presence)
c. `terminals_available` = User has condoned the use of terminals and terminado is installed

Reply:
To be clear, I think there's an actual logic bug here, regardless of which name we're using. ;-)

Reply:
You're absolutely correct.  Thank you for spotting that.

Reply:
Thanks. I think this is in good shape now. :-)